dev/devicelab/lib/framework/adb.dart

```diff
@@ -358,19 +371,37 @@ class AndroidDevice implements Device {
     return wakefulness;
   }
 
+  Future<void> _updateDeviceInfo() async {
+    final String info = await shellEval(
```

Comment:
Uber nit: what about adding try/catch, so if it fails for whatever reason, it doesn't stop the test

Reply:
Well, it's already within a try/catch in the calling function, but I added one around this one too.

Actually, I also changed the other try/catch blocks so that they only catch `Exception`s since otherwise they'll catch everything, including `AssertionError`, `NoSuchMethodError`, and other `Error` subclasses that represent programming errors and not regular failures. I also upgraded some throws in here to throw a `DeviceException` so that they would be caught.